mito-ai/src/components/AttachFileButton.tsx

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) Saga Inc.
+ * Distributed under the terms of the GNU Affero General Public License v3.0 License.
+ */
+
+import React, { useRef, useState } from 'react';
+import IconButton from './IconButton';
+import PaperClipIcon from '../icons/PaperClipIcon';
+import { requestAPI } from '../restAPI/utils';
+
+interface AttachFileButtonProps {
+    onFileUploaded: (fileName: string) => void;
+}
+
+const AttachFileButton: React.FC<AttachFileButtonProps> = ({ onFileUploaded }) => {
+    const fileInputRef = useRef<HTMLInputElement>(null);
+    const [isUploading, setIsUploading] = useState(false);
+
+    // Constants for file handling
+    const MAX_FILE_SIZE = 100 * 1024 * 1024; // 100MB in bytes
+    const CHUNK_SIZE = 50 * 1024 * 1024; // 50MB chunks
+
+    const handleClick = (): void => {
+        // Don't allow clicks if uploading
+        if (isUploading) return;
+
+        // Trigger the hidden file input
+        fileInputRef.current?.click();
+    };
+
+    const handleFileSelect = (event: React.ChangeEvent<HTMLInputElement>): void => {
+        const files = event.target.files;
+        if (!files || files.length === 0) return;
+
+        const file = files[0];
+        if (!file) return;
+
+        // Check file size and handle accordingly
+        if (file.size > MAX_FILE_SIZE) {
+            console.log(`File ${file.name} is larger than 100MB (${(file.size / (1024 * 1024)).toFixed(2)}MB). Splitting into chunks...`);
+            void handleLargeFile(file);
+        } else {
+            // Upload file directly for files <= 100MB
+            void uploadFile(file);
+        }
+    };
+
+    const handleLargeFile = async (file: File): Promise<void> => {
+        setIsUploading(true);
```

Comment:
Maybe we should move the loading handling into the `handleFileSelect` function so it is used for both small and large files.